specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/playbookConfigurations.json

```diff
@@ -0,0 +1,461 @@
+          "$ref": "#/definitions/PlaybookConfigurationProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/Resource"
+        }
+      ]
+    },
+    "PlaybookConfigurationProperties": {
+      "type": "object",
+      "description": "The playbook configuration data.",
+      "properties": {
+        "description": {
+          "type": "string",
+          "description": "The playbook configuration description."
+        },
+        "isEnabled": {
+          "type": "boolean",
+          "description": "Indicates whether the playbook configuration is enabled."
+        },
+        "metadata": {
+          "type": "object",
+          "description": "The metadata of the playbook configuration resource.",
+          "$ref": "#/definitions/PlaybookConfigurationMetadata"
+        },
+        "scopes": {
+          "type": "array",
+          "description": "A collection of the resources scopes on which the playbook configurations logic is applied.",
+          "items": {
+            "type": "string",
+            "description": "The resources scope."
+          }
+        },
+        "sources": {
+          "type": "array",
+          "description": "A collection of the source event types which evaluate the playbook configuration set of rules.",
+          "items": {
+            "$ref": "#/definitions/PlaybookConfigurationSource"
+          }
+        },
+        "actions": {
+          "type": "array",
+          "description": "A collection of the actions which are triggered upon true evaluation of the configured rule sets.",
+          "items": {
+            "$ref": "#/definitions/PlaybookConfigurationAction"
+          }
+        }
+      }
+    },
+    "PlaybookConfigurationMetadata": {
+      "type": "object",
+      "description": "The metadata of the playbook configuration resource.",
+      "properties": {
+        "createdDateTimeUtc": {
+          "readOnly": true,
+          "type": "string",
+          "format": "date-time",
+          "description": "The playbook configuration creation date."
+        },
+        "createdBy": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The AAD object id that created the playbook configuration."
+        },
+        "lastUpdatedDateTimeUtc": {
+          "readOnly": true,
+          "type": "string",
+          "format": "date-time",
+          "description": "The playbook configuration last updated date."
+        },
+        "lastUpdatedBy": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The AAD object id that last updated the playbook configuration."
+        }
+      }
+    },
+    "PlaybookConfigurationSource": {
+      "type": "object",
+      "description": "The source event types which evaluate the playbook configuration set of rules. For example - security alerts and recommended tasks.",
+      "properties": {
+        "eventSource": {
```

Comment:
if this is an enum add that to the swagger including all the possible values

Reply:
for pp, it will be supplied via documentation. For PP, there will be a dedicated API for the valid values.

Reply:
ok

---
In reply to: [276255184](https://github.com/Azure/azure-rest-api-specs/pull/5548#discussion_r276255184) [](ancestors = 276255184)